clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedInternalCommandConsumer.java

```diff
@@ -321,20 +356,35 @@ private String getTopicName() {
         return new HonoTopic(HonoTopic.Type.COMMAND_INTERNAL, adapterInstanceId).toString();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Closes the Kafka admin client and consumer.
+     *
+     * @return A future indicating the outcome of the operation.
+     *         The future will be succeeded once this component is stopped.
+     */
     @Override
     public Future<Void> stop() {
-        retryCreateTopic.set(false);
-        vertx.cancelTimer(retryCreateTopicTimerId);
 
-        if (consumer == null) {
-            return Future.failedFuture("not started");
+        if (lifecycleStatus.isStopped()) {
+            // nothing to do
+            return Future.succeededFuture();
         }
 
-        return CompositeFuture.all(closeAdminClient(), closeConsumer())
+        return lifecycleStatus.runStopAttempt(() -> {
```

Comment:
Always having the extra `isStopped` check outside the `runStopAttempt` method looks a bit inconvenient here.
On the adding hand, changing the semantics of `runStopAttempt` 
from `Executes an attempt to stop the tracked component if it isn't already stopping`
to `Executes an attempt to stop the tracked component if it isn't already stopped or stopping`
would make the methods behaviour rather more consistent and intuitive than less so, FMPOV.

Reply:
I would by fine with that as well :+1: